app/models/az322_contribution.rb

```diff
@@ -45,4 +46,13 @@ def amount=(value)
   end
 
   private
+
+  VALID_MONEY_REGEXP = /\A(\d+)?\.?\d{0,2}\z/.freeze
+  def amount_format
+    return true if amount.blank?
+
+    unless VALID_MONEY_REGEXP.match?(read_attribute_before_type_cast(:amount).to_s)
```

Comment:
`read_attribute_before_type_cast` because the amount gets coerced into decimal type & `to_s` because the regex matching can't happen on valid floats.

Reply:
I am mistaken then! I did try `'100.'.match(<regex here>)` and it matched that as well. I think because the final bit after decimal and the decimal bit are not grouped together.

Reply:
oh, here i was trying to make sure `100.` is considered a "valid" money format (b/c that's what Product/Design wanted to consider valid) -- so the matching should be ok?

Reply:
Oh I gotcha. Sorry getting turned around a little on requirements!